capa/features/freeze/__init__.py

```diff
@@ -46,7 +45,7 @@ class AddressType(str, Enum):
 
 class Address(HashableModel):
     type: AddressType
-    value: Union[int, Tuple[int, int], None]
+    value: Union[int, Tuple[int, int], None] = None
```

Comment:
`Address` shouldn't be instantiated directly (its a super class) so why do we have to provide a default value here?

Reply:
These changes weren't made by me directly, I followed pydantic's migration guide "https://docs.pydantic.dev/latest/migration/". Code was auto refactored. Will look into it .

Reply:
thanks for reverting! if it's easy to explain - great. otherwise, don't worry about it.

Reply:
@williballenthin If `Address` is not instantiated it pydantic validator gives error
```
PS C:\Users\HP\Documents\GitHub\capa> capa '.\tests\data\rd\Practical Malware Analysis Lab 01-01.dll_.json'
Traceback (most recent call last):
  File "C:\Users\HP\AppData\Local\Programs\Python\Python39\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\HP\AppData\Local\Programs\Python\Python39\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\HP\AppData\Local\Programs\Python\Python39\Scripts\capa.exe\__main__.py", line 7, in <module>
  File "C:\Users\HP\Documents\GitHub\capa\capa\main.py", line 1226, in main
    result_doc = capa.render.result_document.ResultDocument.parse_file(args.sample)
  File "C:\Users\HP\AppData\Local\Programs\Python\Python39\lib\site-packages\typing_extensions.py", line 2562, in wrapper
    return __arg(*args, **kwargs)
  File "C:\Users\HP\AppData\Local\Programs\Python\Python39\lib\site-packages\pydantic\main.py", line 1041, in parse_file
    return cls.parse_obj(obj)
  File "C:\Users\HP\AppData\Local\Programs\Python\Python39\lib\site-packages\typing_extensions.py", line 2562, in wrapper
    return __arg(*args, **kwargs)
  File "C:\Users\HP\AppData\Local\Programs\Python\Python39\lib\site-packages\pydantic\main.py", line 961, in parse_obj
    return cls.model_validate(obj)
  File "C:\Users\HP\AppData\Local\Programs\Python\Python39\lib\site-packages\pydantic\main.py", line 496, in model_validate
    return cls.__pydantic_validator__.validate_python(
pydantic_core._pydantic_core.ValidationError: 3 validation errors for ResultDocument
rules.delay execution.matches.0.1.children.1.children.0.locations.0.value
  Field required [type=missing, input_value={'type': 'no address'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.1/v/missing
rules.delay execution.matches.1.1.children.1.children.0.locations.0.value
  Field required [type=missing, input_value={'type': 'no address'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.1/v/missing
rules.send data.matches.0.1.children.0.children.0.locations.0.value
  Field required [type=missing, input_value={'type': 'no address'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.1/v/missing
```

Reply:
thanks for the details and stack trace. i sort of understand the issue - our NO_ADDRESS address doesn't have a value, and pydantic validation isn't happy about this. while we might be able to make a better fix, it would take a bunch of work to track down. so this change is ok with me.